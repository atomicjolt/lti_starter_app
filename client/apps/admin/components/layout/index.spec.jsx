import React from 'react';
import TestRenderer, { act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Index from './index';

const mockStore = configureStore([]);
const store = mockStore({});

describe('layout index', () => {
  let result;
  const location = {
    pathname: '/'
  };
  const text = 'hello';

  beforeEach(() => {
    act(() => {
      result = TestRenderer.create(
        <Provider store={store}>
          <Index location={location}>
            <h1>{text}</h1>
          </Index>
        </Provider>
      );
    });
  });

  it('renders the index', () => {
    expect(result).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('Loads sites and applications', () => {
    // Test that the store dispatched an action to get sites and applications
    const actions = store.getActions();
    expect(!!actions.find((a) => a.type === 'GET_APPLICATIONS')).toEqual(true);
    expect(!!actions.find((a) => a.type === 'GET_SITES')).toEqual(true);
  });
});
