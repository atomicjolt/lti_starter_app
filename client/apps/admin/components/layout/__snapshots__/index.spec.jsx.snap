// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout index matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Index
    getApplications={[Function]}
    getSites={[Function]}
    location={
      Object {
        "pathname": "/",
      }
    }
  >
    <h1>
      hello
    </h1>
  </Index>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Errors) />,
        <h1>
          hello
        </h1>,
      ],
      "className": "app-index",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "hello",
        },
        "ref": null,
        "rendered": "hello",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Errors) />,
          <h1>
            hello
          </h1>,
        ],
        "className": "app-index",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "hello",
          },
          "ref": null,
          "rendered": "hello",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
